EIPS/eip-ens-abi-lookup.md

```diff
@@ -0,0 +1,70 @@
+## Preamble
+
+    EIP: <to be assigned>
+    Title: ENS support for contract ABIs
+    Author: Nick Johnson <nick@ethereum.org>
+    Type: Standard Track
+    Category: ERCs
+    Status: Draft
+    Created: 2017-02-06
+    Requires: 137, 181
+
+
+## Simple Summary
+This EIP proposes a mechanism for storing ABI definitions in ENS, for easy lookup of contract interfaces by callers.
+
+## Abstract
+ABIs are important metadata required for interacting with most contracts. At present, they are typically supplied out-of-band, which adds an additional burden to interacting with contracts, particularly on a one-off basis or where the ABI may be updated over time. The small size of ABIs permits an alternative solution, storing them in ENS, permitting name lookup and ABI discovery via the same process.
+
+ABIs are typically quite compact; the largest in-use ABI we could find, that for the DAO, is 9450 bytes uncompressed JSON, 6920 bytes uncompressed CBOR, and 1128 bytes when the JSON form is compressed with zlib. Further gains on CBOR encoding are possible using a CBOR extension that permits eliminating repeated strings, which feature extensively in ABIs. Most ABIs, however, are far shorter than this, consisting of only a few hundred bytes of uncompressed JSON.
+
+This EIP defines a resolver profile for retrieving contract ABIs, as well as encoding standards for storing ABIs for different applications, allowing the user to select between different representations based on their need for compactness and other considerations such as onchain access.
+
+## Specification
+### ABI encodings
+In order to allow for different tradeoffs between onchain size and accessibility, several ABI encodings are defined. Each ABI encoding is defined by a unique constant with only a single bit set, allowing for the specification of 256 unique encodings in a single uint.
+
+The currently recognised encodings are:
+
+| ID | Description          |
+|----|----------------------|
+| 1  | JSON                 |
+| 2  | zlib-compressed JSON |
+| 4  | CBOR                 |
+| 8  | URI                  |
+
```

Comment:
The URI in this spec points to an ABI object, while the metadata URI is not an ABI object, but contains the ABI too.

Potentially it could be stored if needed, but isn't Swarm a potentially better way to store data? 😉